test/uk/gov/hmrc/merchandiseinbaggage/stubs/MibBackendStub.scala

```diff
@@ -21,52 +21,57 @@ import com.github.tomakehurst.wiremock.client.WireMock.{post, urlPathEqualTo, _}
 
-object MibBackendStub extends MibConfiguration {
+object MibBackendStub extends MibConfiguration with CoreTestData {
 
   val stubbedDeclarationId = DeclarationId("test-mib-be-id")
 
-  def givenDeclarationIsPersistedInBackend(server: WireMockServer, declaration: Declaration): StubMapping =
+  def givenDeclarationIsPersistedInBackend(declaration: Declaration)(implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         post(urlPathEqualTo(s"$declarationsUrl"))
           .withRequestBody(equalToJson(toJson(declaration).toString, true, false))
           .willReturn(aResponse().withStatus(201).withBody(Json.toJson(stubbedDeclarationId).toString)))
 
-  def givenDeclarationIsPersistedInBackend(server: WireMockServer): StubMapping =
+  def givenDeclarationIsPersistedInBackend(implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         post(urlPathEqualTo(s"$declarationsUrl"))
           .willReturn(aResponse().withStatus(201).withBody(Json.toJson(stubbedDeclarationId).toString)))
 
-  def givenPersistedDeclarationIsFound(
-    server: WireMockServer,
-    declaration: Declaration,
-    declarationId: DeclarationId = stubbedDeclarationId): StubMapping =
+  def givenPersistedDeclarationIsFound(declaration: Declaration, declarationId: DeclarationId = stubbedDeclarationId)(
+    implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         get(urlPathEqualTo(s"$declarationsUrl/${declarationId.value}"))
           .willReturn(okJson(Json.toJson(declaration).toString)))
 
-  def givenAPaymentCalculation(server: WireMockServer, request: CalculationRequest, result: CalculationResult): StubMapping =
+  def givenAPaymentCalculation(request: CalculationRequest, result: CalculationResult)(implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         post(urlPathEqualTo(s"$calculationUrl"))
           .withRequestBody(equalToJson(toJson(request).toString, true, false))
           .willReturn(okJson(Json.toJson(result).toString)))
 
-  def givenAPaymentCalculation(server: WireMockServer, result: CalculationResult): StubMapping =
+  def givenAPaymentCalculation(result: CalculationResult)(implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         post(urlPathEqualTo(s"$calculationUrl"))
           .willReturn(okJson(Json.toJson(result).toString)))
 
-  def givenSendEmailsSuccess(server: WireMockServer, declarationId: DeclarationId = stubbedDeclarationId): StubMapping =
+  def givenSendEmailsSuccess(declarationId: DeclarationId = stubbedDeclarationId)(implicit server: WireMockServer): StubMapping =
     server
       .stubFor(
         get(urlPathEqualTo(s"$sendEmailsUrl/${declarationId.value}"))
           .willReturn(aResponse().withStatus(202)))
+
+  def givenEoriIsChecked(eoriNumber: String)(implicit server: WireMockServer): StubMapping =
+    server
+      .stubFor(
+        get(urlPathEqualTo(s"$checkEoriUrl$eoriNumber"))
```

Comment:
its good practise to use full hardcoded expected url so it catches any issues during tests

Reply:
good practise? are you sure??? Hard coded strings hanging around then if you need to make a change you need to go and fish them around the whole code base instead of changing it in one place

Reply:
i mean the full url like `localhost:$wiremockPort/declare-commercial-goods/validate/eori/$eori`  if you use this url in your test this will never be a problem , if you used this , it would have caught that url error during test itself